ztoc/ztoc.go

```diff
@@ -117,44 +109,25 @@ func ExtractFile(r *io.SectionReader, config *FileExtractConfig) ([]byte, error)
 	spanEnd := gzipZinfo.UncompressedOffsetToSpanID(config.UncompressedOffset + config.UncompressedSize)
 	numSpans := spanEnd - spanStart + 1
 
-	var bufSize compression.Offset
-	starts := make([]compression.Offset, numSpans)
-	ends := make([]compression.Offset, numSpans)
+	checkpoints := make([]compression.Offset, numSpans+1)
+	checkpoints[0] = gzipZinfo.SpanIDToStartCompressedOffset(spanStart)
 
 	var i compression.SpanID
 	for i = 0; i < numSpans; i++ {
-		starts[i] = gzipZinfo.SpanIDToCompressedOffset(i + spanStart)
-		if i+spanStart == config.MaxSpanID {
-			ends[i] = config.CompressedArchiveSize - 1
```

Comment:
I don't know why the end compress offset for the MaxSpanID is `CompressedArchiveSize - 1` here but `CompressedArchiveSize` in other places. (why -1 here)

Reply:
I think it should be `CompressedArchiveSize` right? given it's common to use left-inclusive interval (e.g. `slice[start:CompressedArchiveSize]` means `[start, CompressedArchiveSize)`.

Also in gzip_zinfo we define `spanIDToCompressedOffset(i)` as the 1st _complete_ bit belonging to the span `i`. (that's why if `i` `hasBits`, we -1 on the start offset.)

 So the chunk of data for a span is `slice[spanIDToCompressedOffset(i) : spanIDToCompressedOffset(i + 1)]` (e.g. instead of `spanIDToCompressedOffset(i + 1) - 1`) And for the last span its `slice[spanIDToCompressedOffset(i) : CompressedArchiveSize]`

Reply:
Yep, you're right. I think it might still be good to have that test.

Reply:
I believe we get 1 file from each span